Resolver/Elastos-DID-Resolver-Specification_en.md

```diff
@@ -408,3 +383,246 @@ Member optional, contains other information related to the error. The value is d
   }
 }
 ```
+
+## DID Query
+
+DID Query is a method for querying DID documents on the ID side chain. The query scope is all latest DID documents in non-deactivated state, the expired documents are in the scope.
+
+### Query Request
+
+#### method
+
+The string value contains the name of the method to be invoked. For the DID query request, the value is "query".
+
+#### params
+
+The object value, which is the parameter to be included in the call of  `query`  method. The parameters are defined as follows:
+
+- service
+
+  Optional, a string representing the service type. For service type, see the definition of service in the Elastos DID method specification. Support multiple types or "|", and "&", such as: "type1 | type2", "type1 & type2".
+
+- credential
+
+  Optional, a string representing the credential type. For credential type, see the definition of credential in the Elastos Verifiable Claims Specification. Support multiple types or "|", and "&", such as: "type1 | type2", "type1 & type2".
+
+- query
+
+  Optional, query conditions in MongoDB query syntax.
```

Comment:
Maybe we should specify that this can only use mongo's "find/match" format, not other formats for example the ones used for aggregation pipelines, etc